src/main/java/com/boardgamesworld/bgrental/boardgame/infrastructure/api/BoardGameController.java

```diff
@@ -0,0 +1,124 @@
+package com.boardgamesworld.bgrental.boardgame.infrastructure.api;
+
+import com.boardgamesworld.bgrental.boardgame.domain.BoardGame;
+import com.boardgamesworld.bgrental.boardgame.domain.BoardGameFacade;
+import com.boardgamesworld.bgrental.boardgame.domain.InvalidBoardGameException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/boardgames")
+public class BoardGameController {
+
+    private BoardGameFacade boardGameFacade;
+    private Logger logger = LoggerFactory.getLogger(BoardGameController.class);
+
+    public BoardGameController(BoardGameFacade boardGameFacade) {
+        this.boardGameFacade = boardGameFacade;
+    }
+
+    @GetMapping
+    public ResponseEntity<List<BoardGameDto>> getAllBoardGames() {
+        List<BoardGame> boardGames = boardGameFacade.getAllBoardGames();
+        List<BoardGameDto> boardGamesDto;
+
+        if (boardGames != null) {
+            boardGamesDto = boardGames.stream()
+                    .map(boardGame -> new BoardGameDto(
```

Comment:
tutaj mo≥że być metoda, ktora przyjmuje BoardGame i zwraca BoardGameDto, wtedy bedize tu tylko
`map(this::toDto)`